### PR TITLE
gh11701 add C_BPartner_RelatedRecord1 table and C_BPartner_EditableAttribute1

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/11701

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecord.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import de.metas.util.Check;
+import de.metas.util.lang.RepoIdAware;
+import lombok.Value;
+
+@Value
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY, getterVisibility = JsonAutoDetect.Visibility.NONE, isGetterVisibility = JsonAutoDetect.Visibility.NONE, setterVisibility = JsonAutoDetect.Visibility.NONE)
+public class BPartnerRelatedRecord implements RepoIdAware
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecord.java:34]: rename it to `BPartnerRelatedRecordId`

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/related/service/adapter/BPRelatedRecords1Adapter.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	protected IQuery<I_C_BPartner_RelatedRecord1> queryRecords(final @NonNull BPartnerId bpartnerId)
+	{
+		return queryBL.createQueryBuilder(I_C_BPartner_RelatedRecord1.class)
+				.addEqualsFilter(I_C_BPartner_RelatedRecord1.COLUMNNAME_C_BPartner_ID, bpartnerId)
+				.orderBy(I_C_BPartner_RelatedRecord1.COLUMNNAME_Record_ID)
+				.create();
+	}
+
+	@Override
+	protected BPartnerRelatedRecord extractRecord(final @NonNull I_C_BPartner_RelatedRecord1 record)
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/related/service/adapter/BPRelatedRecords1Adapter.java:44]: extractRecordId

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecords.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.NonNull;
+import lombok.Value;
+
+@Value
+@Builder
+public class BPartnerRelatedRecords
+{
+	@NonNull
+	@Builder.Default
+	ImmutableSet<BPartnerRelatedRecord> relatedRecords1 = ImmutableSet.of();
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecords.java:36]: relatedRecordIds1

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/lookup/LabelsLookup.java [hunk 4]
```diff
@@ -250,7 +257,8 @@ public LookupValue retrieveLookupValueById(final @NonNull LookupDataSourceContex
 	{
 		return LookupDataSourceContext.builder(tableName)
 				.setRequiredParameters(parameters)
-				.requiresAD_Language();
+				.requiresAD_Language()
+				.requiresUserRolePermissionsKey();
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/lookup/LabelsLookup.java:261]: why is `requiresUserRolePermissionsKey` needed?

Comment (teosarca) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/lookup/LabelsLookup.java:261]: IMHO it's not needed anymore.

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java [hunk 5]
```diff
@@ -312,6 +312,11 @@ public static void reset(final boolean finalCall)
 	 */
 	public static final String DYNATTR_TabNo = "TabNo";
 
+	/**
+	 * To be used when setting the current user ID in context
+	 */
+	public static final String DYNATTR_AD_User_ID = "AD_User_ID";
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java:318]: IMHO this is not needed.

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java (hunk 5 above)
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java]: removed

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/lookup/LabelsLookup.java (hunk 4 above)
Comment (adi-stefan) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/lookup/LabelsLookup.java]: true, removed.

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecords.java (hunk 3 above)
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecords.java]: renamed

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/related/service/adapter/BPRelatedRecords1Adapter.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/related/service/adapter/BPRelatedRecords1Adapter.java]: renamed.

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecord.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/attributes/BPartnerRelatedRecord.java]: renamed.

**Pull request merged**